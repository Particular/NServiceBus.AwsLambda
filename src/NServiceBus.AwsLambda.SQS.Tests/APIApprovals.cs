namespace NServiceBus.AwsLambda.Tests
{
    using NUnit.Framework;
    using Particular.Approvals;
    using PublicApiGenerator;

    [TestFixture]
    public class APIApprovals
    {
        [Test]
        public void Approve()
        {
            var options = new ApiGeneratorOptions
            {
                ExcludeAttributes = new[] {"System.Runtime.Versioning.TargetFrameworkAttribute"}
            };
            var publicApi = typeof(AwsLambdaSQSEndpoint).Assembly.GeneratePublicApi(options);
            Approver.Verify(publicApi);
        }
    }
}