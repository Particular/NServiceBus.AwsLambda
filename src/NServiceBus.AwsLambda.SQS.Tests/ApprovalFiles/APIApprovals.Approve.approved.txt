namespace NServiceBus
{
    public class AwsLambdaSQSEndpoint : NServiceBus.Serverless.ServerlessEndpoint<Amazon.Lambda.Core.ILambdaContext, NServiceBus.SQSTriggeredEndpointConfiguration>
    {
        public AwsLambdaSQSEndpoint(System.Func<Amazon.Lambda.Core.ILambdaContext, NServiceBus.SQSTriggeredEndpointConfiguration> configurationFactory) { }
        protected override System.Threading.Tasks.Task Initialize(NServiceBus.SQSTriggeredEndpointConfiguration configuration) { }
        public System.Threading.Tasks.Task Process(Amazon.Lambda.SQSEvents.SQSEvent @event, Amazon.Lambda.Core.ILambdaContext lambdaContext) { }
    }
    public class SQSTriggeredEndpointConfiguration : NServiceBus.Serverless.ServerlessEndpointConfiguration
    {
        public SQSTriggeredEndpointConfiguration(string endpointName) { }
        public NServiceBus.TransportExtensions<NServiceBus.SqsTransport> Transport { get; }
    }
}